File: src/main/java/net/softbell/bsh/controller/view/NodeView.java
package net.softbell.bsh.controller.view;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import lombok.AllArgsConstructor;
import net.softbell.bsh.domain.entity.Node;
import net.softbell.bsh.domain.entity.NodeItem;
import net.softbell.bsh.domain.entity.NodeItemHistory;
import net.softbell.bsh.iot.service.v1.IotNodeServiceV1;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 노드 뷰 컨트롤러
 */
@AllArgsConstructor
@Controller
@RequestMapping("/node")
public class NodeView
{
	// Global Field
	private final String G_BASE_PATH = "services/advance";
	private final IotNodeServiceV1 iotNodeService;
	
	@GetMapping()
    public String dispIndex(Model model,
    						@RequestParam(value = "page", required = false, defaultValue = "1")int intPage,
    						@RequestParam(value = "count", required = false, defaultValue = "100")int intCount)
	{
		// Field
		Page<Node> pageNode;
		
		// Init
		pageNode = iotNodeService.getAllNodes(intPage, intCount);
		
		// Process
		model.addAttribute("listNode", pageNode);
		
		// Return
        return G_BASE_PATH + "/NodeList";
    }
	
	@GetMapping("/{id}")
    public String dispNode(Model model, @PathVariable("id")int intNodeId)
	{
		// Field
		Node node;
		
		// Init
		node = iotNodeService.getNode(intNodeId);
		
		// Process
		model.addAttribute("nodeInfo", node);
		
		// Return
        return G_BASE_PATH + "/NodeInfo";
    }
	
	@GetMapping("/item/{id}")
    public String dispNodeItemHistory(Model model, @PathVariable("id")int intNodeItemId)
	{
		// Field
		List<NodeItemHistory> pageNodeItemHistory;
		NodeItem nodeItem;
		
		// Init
		pageNodeItemHistory = iotNodeService.getNodeItemHistory(intNodeItemId);
		nodeItem = iotNodeService.getNodeItem(intNodeItemId);
		
		// Process
		model.addAttribute("listNodeItemHistory", pageNodeItemHistory);
		model.addAttribute("nodeItem", nodeItem);
		
		// Return
        return G_BASE_PATH + "/nodeItemInfo";
    }
}

File: src/main/kotlin/net/softbell/bsh/controller/view/advance/NodeView.kt
package net.softbell.bsh.controller.view.advance

import net.softbell.bsh.domain.GroupRole
import net.softbell.bsh.dto.view.advance.NodeInfoCardDto
import net.softbell.bsh.dto.view.advance.NodeItemHistoryCardDto
import net.softbell.bsh.iot.service.v1.IotNodeServiceV1
import net.softbell.bsh.service.CenterService
import net.softbell.bsh.service.ViewDtoConverterService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam

/**
 * @author : Bell(bell@softbell.net)
 * @description : 노드 뷰 컨트롤러
 */
@Controller
@RequestMapping("/node")
class NodeView {
    // Global Field
    private val G_BASE_PATH: String = "services/advance"
    private val G_INDEX_REDIRECT_URL: String = "redirect:/"

    @Autowired private lateinit var viewDtoConverterService: ViewDtoConverterService
    @Autowired private lateinit var iotNodeService: IotNodeServiceV1
    @Autowired private lateinit var centerService: CenterService

    @GetMapping
    fun dispIndex(model: Model, auth: Authentication,
                  @RequestParam(value = "page", required = false, defaultValue = "1") intPage: Int,
                  @RequestParam(value = "count", required = false, defaultValue = "100") intCount: Int): String {
        // Exception
        if (centerService.setting.iotNode != 1.toByte())
            return G_INDEX_REDIRECT_URL

        // Init
        val listNode = iotNodeService.getAllNodes(auth, GroupRole.MANUAL_CONTROL)

        // Process
        model.addAttribute("listCardNodes", viewDtoConverterService.convNodeSummaryCards(listNode))

        // Return
        return "$G_BASE_PATH/NodeList"
    }

    @GetMapping("/{id}")
    fun dispNode(model: Model, @PathVariable("id") intNodeId: Long): String {
        // Exception
        if (centerService.setting.iotNode != 1.toByte())
            return G_INDEX_REDIRECT_URL

        // Init
        val node = iotNodeService.getNode(intNodeId)

        // Process
        model.addAttribute("cardNodeInfo", node?.let { NodeInfoCardDto(it) })
        if (node != null) {
            model.addAttribute("listCardNodeItems", viewDtoConverterService.convNodeItemCards(node.nodeItems))
        }

        // Return
        return "$G_BASE_PATH/NodeInfo"
    }

    @GetMapping("/item/{id}")
    fun dispNodeItemHistory(model: Model, @PathVariable("id") intNodeItemId: Long): String {
        // Exception
        if (centerService.setting.iotNode != 1.toByte())
            return G_INDEX_REDIRECT_URL

        // Init
        val nodeItem = iotNodeService.getNodeItem(intNodeItemId)
        val listNodeItemHistory = iotNodeService.getNodeItemHistory(intNodeItemId)

        // Process
        model.addAttribute("cardNodeItemHistory", nodeItem?.let { NodeItemHistoryCardDto(it, listNodeItemHistory) })

        // Return
        return "$G_BASE_PATH/NodeItemInfo"
    }
}